examples/angular/src/pages/ui/components/authenticator/custom-slots/custom-slots.component.html

```diff
@@ -0,0 +1,77 @@
+<amplify-authenticator>
+  <ng-template amplifySlot="header">
+    <div style="padding: var(--amplify-space-large); text-align: center">
+      <img
+        class="amplify-image"
+        alt="Amplify logo"
+        src="https://docs.amplify.aws/assets/logo-dark.svg"
+      />
+    </div>
+  </ng-template>
+
+  <ng-template amplifySlot="sign-in-header">
+    <h3
+      class="amplify-heading"
+      style="padding: var(--amplify-space-xl) 0 0 var(--amplify-space-xl)"
+    >
+      Sign in to your account
+    </h3>
+  </ng-template>
+
+  <ng-template amplifySlot="sign-in-footer" let-authenticator="authenticator">
+    <div style="text-align: center">
+      <button
+        (click)="authenticator.toResetPassword()"
```

Comment:
I'm trying to figure out the best way to get this `(click)` working.

The flaw with slots is if they have a limited subset of functionality exposed to them. For example, we can't assume that a `sign-in-footer` will **only** need `toResetPassword`.

Ideally, exposing `authenticator.*` is best.

Reply:
Oops, I think I forgot to add back context to each template in #598.

https://github.com/aws-amplify/amplify-ui/blob/ac4eceeeb009dfd427622ef6babaaed189fe1591/packages/angular/projects/ui-angular/src/lib/components/authenticator/components/sign-in/sign-in.component.html#L34

should be

```diff
- <amplify-slot name="sign-in-footer"> </amplify-slot> 
+ <amplify-slot name="sign-in-footer" [context]="context"> </amplify-slot> 
```

where the `context` is picked up from this getter:

https://github.com/aws-amplify/amplify-ui/blob/ac4eceeeb009dfd427622ef6babaaed189fe1591/packages/angular/projects/ui-angular/src/lib/components/authenticator/components/sign-in/sign-in.component.ts#L24-L28

Reply:
> exposing authenticator.* is best.

Agreed, I was thinking of the same thing! `authenticator.*` is exposed from `useAuthenticator` anyway so it's not exposing anymore risks. I can make that change tomorrow. 

For this PR, you can pick up the above change and use 

```html
  <ng-template amplifySlot="sign-in-footer" let-toResetPassword="toResetPassword">
    <div style="text-align: center">
      <button
        (click)="toResetPassword()"
```
to unblock the pr.

Reply:
@wlee221 I'm took a different approach, I brought in the authenticator service and added it into a getter. Then accessed that in the `(click)` handler.

```
  get auth() {
    const { toResetPassword, toSignIn } = this.authenticator;
    return { toResetPassword, toSignIn };
  }
```

```

(click)="auth.toSignIn()"
```

Reply:
Maybe this is OK for now? Otherwise we need to figure out how to pass the `context` into the slot... Which isn't setup right now.

Reply:
I'll add in the `context` for `sign-up-footer` and `sign-in-footer` for now so this should work 

```
  <ng-template amplifySlot="sign-in-footer" let-toResetPassword="toResetPassword">
```

But at some point we'll need to add back `context` in on all slots.